cli/popper/runner_host.py

```diff
@@ -251,3 +256,189 @@ def _update_with_engine_config(self, container_args):
         for k, v in update_with.items():
             if k not in container_args.keys():
                 container_args[k] = update_with[k]
+
+
+class SingularityRunner(StepRunner):
+    """Runs steps in singularity on the local machine."""
+    lock = threading.Lock()
+
+    def __init__(self, **kw):
+        super(SingularityRunner, self).__init__(**kw)
+
+        self._spawned_containers = set()
+        self._s = None
+
+        if self._config.reuse:
+            log.fail('Reuse not supported for SingularityRunner.')
+
+        self._s = spython.main.Client
+        self._s.quiet = True
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        self._spawned_containers = set()
+
+    def run(self, step):
+        self._setup_singularity_cache()
+        cid = pu.sanitized_name(step['name'], self._config.wid) + '.sif'
+        self._container = os.path.join(self._singularity_cache, cid)
+
+        exists = os.path.exists(self._container)
+        if exists and not self._config.reuse and not self._config.dry_run and not self._config.skip_pull:
```

Comment:
Since `reuse` is not supported in singularity, I have already removed the condition of `not self._config.reuse`. Now the line size is within PEP8 limits.